OrbitGl/App.cpp

```diff
@@ -329,7 +329,47 @@ void OrbitApp::Disassemble(int32_t pid, const Function& function) {
     disasm.Disassemble(reinterpret_cast<const uint8_t*>(memory.data()),
                        memory.size(), function.GetVirtualAddress(),
                        Capture::GTargetProcess->GetIs64Bit());
-    SendDisassemblyToUi(disasm.GetResult());
+    if (!sampling_report_ || !sampling_report_->GetProfiler()) {
+      SendDisassemblyToUi(disasm.GetResult());
+      return;
+    }
+    std::shared_ptr<SamplingProfiler> profiler =
+        sampling_report_->GetProfiler();
+    unsigned int count_of_function =
+        profiler->GetCountOfFunction(function.GetVirtualAddress());
+    if (count_of_function == 0) {
+      SendDisassemblyToUi(disasm.GetResult());
+      return;
+    }
+
+    const std::function<const double(size_t)> line_to_hits =
+        [profiler, disasm, count_of_function](size_t line) {
+          LOG("Hits of line: %d", line);
+          uint64_t address = disasm.GetAddressAtLine(line);
+          if (address == 0) {
+            return 0.0;
+          }
+
+          const ThreadSampleData& data = profiler->GetSummary();
```

Comment:
Nit: data is only needed in the while loop, so you could reduce the scope of this variable by moving it down.

Reply:
Sure. Done.